### PR TITLE
An erc20 implement for Java contract

### PR DESCRIPTION
**Pull request by Tri-stone**
## Description

An erc20 implement for Java contract.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

It has been tested.

### CODE REVIEW

File: core/contractsdk/java/example/erc20/src/main/java/com/baidu/xuper/Erc20.java [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+        String toKey = BALANCEPRE + to;
+        byte[] toBalanceByte = ctx.getObject(toKey.getBytes());
+        BigInteger toBalance;
+        if (toBalanceByte == null) {
+            toBalance = BigInteger.valueOf(0);
+        } else {
+            toBalance = new BigInteger(toBalanceByte);
+        }
+
+        BigInteger fromBalanceNow = fromBalance.subtract(tokenAmount);
```
Comment (icexin) [inline core/contractsdk/java/example/erc20/src/main/java/com/baidu/xuper/Erc20.java:263]: The from party also needs to check whether the balance is sufficient

Comment (Tri-stone) [reply core/contractsdk/java/example/erc20/src/main/java/com/baidu/xuper/Erc20.java:268]: fixed

File: core/contractsdk/java/example/erc20/src/main/java/com/baidu/xuper/Erc20.java [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+            return Response.error("missing to");
+        }
+        String to = new String(toByte);
+
+        byte[] tokenByte = ctx.args().get("token");
+        if (tokenByte == null) {
+            return Response.error("missing token");
+        }
+        String tokenAmountStr = new String(tokenByte);
+        BigInteger tokenAmount = new BigInteger(tokenAmountStr);
```
Comment (yucaowang) [inline core/contractsdk/java/example/erc20/src/main/java/com/baidu/xuper/Erc20.java:173]: What if the token is negative?

Comment (Tri-stone) [reply core/contractsdk/java/example/erc20/src/main/java/com/baidu/xuper/Erc20.java:179]: It's fixed. Token amount could not be negative now.

**Pull request merged**